sphinx + Travis CI builds failing

Travis build reports for #473 and #442 include the following error:

```
$ make $BUILD_CMD
sphinx-build -b html -W -d _build/doctrees   . _build/html
Running Sphinx v1.5
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 166 source files that are out of date
updating environment: 166 added, 0 changed, 0 removed
reading sources... [100%] troubleshooting
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [  0%] glossary
Exception occurred:
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/sphinx/environment/managers/toctree.py", line 273, in _entries_from_toctree
    title = clean_astext(self.titles[ref])
AttributeError: 'Toctree' object has no attribute 'titles'
The full traceback has been saved in /tmp/sphinx-err-EyDmTM.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [html] Error 1
```

Can't think what's changed that would cause this (other than an update to the `ubuntu:latest` docker image.  `make html` works fine from conda envs on my local machine and on sharc.  Investigating.

--- a/index.rst
+++ b/index.rst
@@ -31,7 +31,6 @@ Take a look at the `Research Software Engineering <http://rse.shef.ac.uk/>`_ web
    :maxdepth: 1
    :hidden:
 
-   self
    hpc/index
    iceberg/index
    sharc/index

--- a/parallel/index.rst
+++ b/parallel/index.rst
@@ -13,7 +13,6 @@ If you need advice on how to parallelise your workflow, please contact the `Rese
    :maxdepth: 1
    :hidden:
 
-   self
    JobArray
    SMP
    MPI

